include/internal/rcu.h

```diff
@@ -0,0 +1,43 @@
+/*
+ *
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+
+
+#ifndef OPENSSL_RCU_H
+# define OPENSSL_RCU_H
+# pragma once
+
+struct rcu_qp;
+
+typedef void (*rcu_cb_fn)(void *data);
+
+struct rcu_cb_item {
```

Comment:
This doesn't need to be public now, does it?

Reply:
well, I take that back-ish.  Its not strictly speaking public, since its an internal header, and I could make it completely private, but doing so would require defining the structure in both the threads_pthread.c and threads_win.c files.  If thats not objectionable, I'll do so, but I like to keep it defined in a single location if I can

Reply:
For this kind of declarations we usually add a local header placed into the same directory as the .c files. Also I wonder if we should move the threads_*.c files into the thread subdirectory. There is a plenty of random stuff in crypto so that it IMO desperately needs some cleanup.